src/sql/src/normalize.rs

```diff
@@ -107,23 +109,60 @@ pub fn op(op: &Op) -> Result<&str, PlanError> {
     Ok(&op.op)
 }
 
+#[derive(Debug)]
+pub enum SqlMaybeValueId {
+    Value(Value),
+    Secret(GlobalId),
+}
+
+impl fmt::Display for SqlMaybeValueId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SqlMaybeValueId::Value(v) => write!(f, "{}", v),
+            SqlMaybeValueId::Secret(id) => write!(f, "{}", id),
+        }
+    }
+}
+
+impl SqlMaybeValueId {
```

Comment:
hmm -- I'm not sold on implementing `TryInto` because it's not clear that `SqlMaybeValueId` -> `Value` ought to be the default conversion -- rather than like `SqlMaybeValueId` -> `StringOrGlobalId`.

I don't mind dumping the `try_as_value` impl and just use `try_into_value` everywhere if you think having both is too much.

How does that sound?